contracts/contracts/multisig/Multisig.cairo

```diff
@@ -326,52 +308,45 @@ end
 #
 
 @constructor
-func constructor{
-        syscall_ptr : felt*,
-        pedersen_ptr : HashBuiltin*,
-        range_check_ptr,
-    }(
-        owners_len : felt,
-        owners : felt*,
-        confirmations_required : felt,
-    ):
+func constructor{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    owners_len : felt, owners : felt*, confirmations_required : felt
+):
+    const lower_bound = 1
+
     with_attr error_message("invalid number of required confirmations"):
-        assert_le(confirmations_required, owners_len)
+        assert_in_range(confirmations_required, lower_bound, owners_len + 1)
```

Comment:
Here we force at least 1 confirmation, but we don't force at least 1 owner. This means the ownership of the multisig can be burned (set to 0 owners), which could be a desired features for some use cases.

Thinking out loud if we are ok with the design 🤔  cc @archseer

Reply:
Serum multisig does checks the other way around: `owners_len` must be at least equal to threshold or more: https://github.com/project-serum/multisig/blob/d6fbe734dc4026b9c8fdd2f9e78904112150fa74/programs/multisig/src/lib.rs#L127-L129=

Reply:
@krebernisak as @archseer pointed out all implementation I saw use such logic. Also its not clear for me what is the use case 0 owners and how at least 1 confirmation aligns with this? We want multisig to execute any transaction in that case? Looks like confirmations in that case pointless. 

That also would require a lot of change in the code

Reply:
Let's do it like Serum does it:

```rust
        assert_unique_owners(&owners)?;
        require!(!owners.is_empty(), InvalidOwnersLen);

        let multisig = &mut ctx.accounts.multisig;

        if (owners.len() as u64) < multisig.threshold {
            multisig.threshold = owners.len() as u64;
        }
```

tldr; Owners can not be empty, and threshold size is bound to owners size.

Reply:
In constructor they use such logic:
```
        assert_unique_owners(&owners)?;
        require!(
            threshold > 0 && threshold <= owners.len() as u64,
            InvalidThreshold
        );
        require!(!owners.is_empty(), InvalidOwnersLen);
```

which is what is done here. I think constructor shalln't be as forgiving as set_owners if invalid threshold was passed then we panic

Reply:
Let's match this in the constructor then. Calling `set_owners` with 0 owners can make sense if you're trying to burn the multisig, but constructing a multisig with 0 owners is almost 100% a bug because the contract is useless on deploy.

So: can't construct with 0 owners, but can `set_owners` to none.